actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -21,8 +21,14 @@ def default_render(*args)
         if block_given?
           yield(*args)
         else
-          logger.info "No template found for #{self.class.name}\##{action_name}, rendering head :no_content" if logger
-          super
+          if no_templates?
+            logger.info "No template found for #{self.class.name}\##{action_name}, rendering head :no_content" if logger
+            super
+          else
+            format = request.format.to_sym.upcase if request.format
```

Comment:
Also, what is the case when `request.format` is falsey and what happens in the log message below in that case?